base.go

```diff
@@ -316,18 +316,27 @@ func getOCI(o BaseLayerOpts) error {
 	return extractOutput(o)
 }
 
-func getBuilt(o BaseLayerOpts, sf *Stackerfile) error {
+func getBuilt(o BaseLayerOpts, sfs []*Stackerfile) error {
 	// We need to copy any base OCI layers to the output dir, since they
 	// may not have been copied before and the final `umoci repack` expects
 	// them to be there.
 	base := o.Layer
 	for {
-		var ok bool
-		base, ok = sf.Get(base.From.Tag)
+		// Iterate through base layers until we find the first one with is not BuiltType
+		ok := false
+		for _, sf := range sfs {
```

Comment:
This is essentially the same code as in the build cache's lookupLayerDefinition(); I wonder if you can make a typedef for a list of stackerfiles and just have the lookup as a method on that?

Reply:
Done, but with a map, since we were already using a map in other places, and right now we have consistent usage.